src/ArrayManifold.jl

```diff
@@ -130,6 +130,80 @@ function exp!(M::ArrayManifold, y, x, v; kwargs...)
     return y
 end
 
+function get_basis(
+    M::ArrayManifold,
+    p,
+    B::CachedBasis{<:AbstractOrthonormalBasis{ℝ},T,ℝ},
+) where {T<:AbstractVector}
+    bvectors = get_vectors(M, p, B)
+    N = length(bvectors)
+    M_dim = manifold_dimension(M)
+    if N != M_dim
+
+        throw(ArgumentError("Incorrect number of basis vectors; expected: $M_dim, given: $N"))
+    end
+    for i = 1:N
+        Xi_norm = norm(M, p, bvectors[i])
+        if !isapprox(Xi_norm, 1)
+            throw(ArgumentError("vector number $i is not normalized (norm = $Xi_norm)"))
+        end
+        for j = i+1:N
+            dot_val = real(inner(M, p, bvectors[i], bvectors[j]))
+            if !isapprox(dot_val, 0; atol = eps(eltype(p)))
+                throw(ArgumentError("vectors number $i and $j are not orthonormal (inner product = $dot_val)"))
+            end
+        end
+    end
+    return B
+end
+
+function get_coordinates(M::ArrayManifold, p, X, B::AbstractBasis; kwargs...)
+    is_tangent_vector(M, p, X, true; kwargs...)
+    return get_coordinates(M.manifold, p, X, B)
+end
+for BT in DISAMBIGUATION_BASIS_TYPES
```

Comment:
I thought the new scheme might make it possible not to have these? Though I first have to check carefully ;)

Reply:
No, this is still necessary. But the new scheme removed some other method ambiguities, so it has some benefits.